archive_upload/handlers/dsmc_handlers.py

```diff
@@ -835,39 +882,57 @@ def _create_tarball_cmd(tarball_name, path_to_archive, exclude_from_tarball):
                "--file={} " \
                "{} " \
                ".".format(
-            path_to_archive,
-            tarball_name,
-            tarball_name,
-            exclude_patterns
-        )
+                   path_to_archive,
+                   tarball_name,
+                   tarball_name,
+                   exclude_patterns
+               )
 
     @staticmethod
-    def _remove_tarballed_files_cmd(path_to_archive, tarball_name):
-        return "cd {} && " \
-               "tar " \
-               "--list " \
-               "--file={} |" \
+    def _remove_tarballed_files_cmd(tarball_list_file):
+        # list all non-directory paths, filter them against the tarball contents and
+        # remove the paths that have been added to the tarball
+        return "find . " \
+               "-depth " \
+               "-not -type d |" \
+               "grep " \
+               "-x " \
+               "-f {} |" \
                "xargs " \
                "-n1 " \
-               "rm -f".format(
-            path_to_archive,
-            tarball_name
-        )
+               "-I% " \
+               "rm -f \"%\"".format(
+                   tarball_list_file
+               )
```

Comment:
This command is a bit complex, maybe you could add a short comment on each line to briefly describe what each option does (I'm thinking about `-x`, `-f`, `-n1` and `-I%` in particular).

(same comment for the command below)